tests/sentry/integrations/github/test_integration.py

```diff
@@ -880,3 +880,58 @@ def test_get_trees_for_org_falls_back_to_cache_once_MAX_CONNECTION_ERRORS_is_hit
                     ("baz", "master", []),
                 ]
             )
+
+    @responses.activate
+    def test_source_url_matches(self):
+        installation = self.get_installation_helper()
+
+        test_cases = [
+            (
+                "https://github.com/Test-Organization/sentry/blob/master/src/sentry/integrations/github/integration.py",
+                True,
+            ),
+            (
+                "https://notgithub.com/Test-Organization/sentry/blob/master/src/sentry/integrations/github/integration.py",
+                False,
+            ),
+            ("https://jianyuan.io", False),
+        ]
+        for source_url, matches in test_cases:
+            assert installation.source_url_matches(source_url) == matches
+
+    @responses.activate
+    def test_extract_branch_from_source_url(self):
+        installation = self.get_installation_helper()
+        integration = Integration.objects.get(provider=self.provider.key)
+        repo = Repository.objects.create(
+            organization_id=self.organization.id,
+            name="Test-Organization/repo",
+            url="https://github.com/Test-Organization/repo",
+            provider="integrations:github",
+            external_id=123,
+            config={"name": "Test-Organization/repo"},
+            integration_id=integration.id,
+        )
```

Comment:
Can you wrap this with `with assume_test_silo_mode(SiloMode.REGION):` (see [example here](https://github.com/getsentry/sentry/blob/master/tests/sentry/integrations/github/test_integration.py#L230-L238)). This should resolve the failing test.

Reply:
https://github.com/getsentry/sentry/blob/master/tests/sentry/integrations/github/test_integration.py#L230-L238

Reply:
For context, we're splitting up our codebase + database to deploy in separate silos so we can differentiate between Sentry user data vs Sentry users' user data.

As such, some models are not available everywhere and this wrapper helps the tests pretend that they were created in the correct silo.

Reply:
@leedongwei Done!